mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1636,6 +1639,84 @@ private static ReadPreference getReadPreference(AssociationContext associationCo
 		return associationContext.getAssociationTypeContext().getOptionsContext().getUnique( ReadPreferenceOption.class );
 	}
 
+
+	@Override
+	public boolean supportsNamedPosition() {
+		return false;
+	}
+
+	/**
+	 * call stored procedure
+	 * @param storedProcedureName name of stored procedure.
+	 * @param params - array with values
+	 * @param tupleContext the tuple context
+	 *@see <a href="https://stackoverflow.com/questions/32480060/call-mongodb-function-from-java"> example</a>
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure(String storedProcedureName, Object[] params,
+			TupleContext tupleContext) {
+		StringBuffer commandLine = new StringBuffer( storedProcedureName ).append( "(" );
+		for ( Object param : params ) {
+			if ( param instanceof String ) {
+				commandLine.append( '\'' ).append( param.toString() ).append( '\'' );
+			}
+			else {
+				commandLine.append( param.toString() );
+			}
+			commandLine.append( "," );
+		}
+		commandLine.setLength( commandLine.length() - 1 );
+		commandLine.append( ")" );
+		Document result = provider.getDatabase().runCommand( new Document( "$eval", commandLine.toString() ) );
+		Double ok = result.getDouble( "ok" );
+		if ( ok.intValue() != 1 ) {
+			// return not ok! throw exception?
```

Comment:
I suppose you should do something here?

Reply:
Hi @gsmet !
MongoDB returns result of execution in parameter "ok". Type of value is double. If execution is success ... "ok"=1.0.

In this way ... I think Should I throw exception if "ok"<>1.0 ?

Reply:
I guess it would be nice to have a test

Reply:
Yes, let's throw an exception, but please, add a test .

Reply:
okey @DavideD